lib/matplotlib/axes/_axes.py

```diff
@@ -88,6 +88,26 @@ def _plot_args_replacer(args, data):
                          "multiple plotting calls instead.")
 
 
+class ViolinStatFunc:
+    """
+    The :class:`ViolinStatFunc` contains:
+    1) a callable whose first argument is compulsory and is a 1-d list of data
+    that is used to plot the violin. This first argument is not required to be
+    specified.
+    2) an alias for this callable. When violinplot outputs the dictionary of
+    artists, this alias is used to identify the artist object corresponding to
+    this callable
+    3) a list of additional arguments. This list does not contain the
+    aforementioned compulsory 1-d list of data.
+    """
+    def __init__(self, func_callable, **kargs):
+        self.func_callable = func_callable
+        self.alias = kargs.pop('alias', func_callable.__name__)
+        self.optional_args = kargs.pop('args', [])
+        if not isinstance(self.optional_args, list):
```

Comment:
Any iterable should be fine. Or, at the very least, tuples should be valid as well.

So something like:

```python
if np.isscalar(self.optional_args):
    self.optional_args = list(self.optional_args)
```